tools/codegen/gen-config/gen_config_model.go

```diff
@@ -0,0 +1,59 @@
+package genconfig
+
+type Config struct {
+	Resources map[string]Resource `yaml:"resources"`
+	// DataSources map[string]DataSource `yaml:"data_sources"`
+}
+
+type Resource struct {
+	Create        *APIOperation `yaml:"create"`
+	Read          *APIOperation `yaml:"read"`
+	Update        *APIOperation `yaml:"update"`
+	Delete        *APIOperation `yaml:"delete"`
+	SchemaOptions SchemaOptions `yaml:"schema"`
+}
+
+// type DataSource struct {
```

Comment:
will remove data source aspects for now since this isn't part of scope at this point